### ISSUE TITLE
Cannot run demo in pycharm(2019.1) in windows 10.

### ISSUE THREAD

**Issue by 179416326** [knowledgedefinednetworking/demo-routenet#4]
Hi, i have successfully run this demo in Ubuntu, but when i ran it in pycharm(2019.1 version) in windows 10, it cannot start training, i attached the information in pycharm terminal in the fowllowing .txt file. The NVIDIA GNU TOOL KIT(CUDA+CUDNN) have been installed， please have a look at the attached .txt file, thanks.
[demoroutenet_pycharm.TXT](https://github.com/knowledgedefinednetworking/demo-routenet/files/4364911/demoroutenet_pycharm.TXT)

**Comment by jsuarezv** [comment]
Hello,
We tested the code in Windows 10 with PyCharm and it worked. Probably, you have a problem with the library dependencies or the CUDA distribution. Please make sure you are using the same versions of the libraries listed in the "requirements.txt" file.

Regards,
José

**Comment by 179416326** [comment]
Thanks for replying in a short time, i checked the installed libraries for "requirements.txt" file once again, it matches requirements.txt. For my test in windows, i installed CUDA(10.0 ver.) and CUDNN(7.6.4 ver.), so i wonder if there is something different from your CUDA&CUDNN version.  Thanks, Wayne

**Comment by jsuarezv** [comment]
Hello Wayne,
Please check you are using TensorFlow v1.14. In principle, with this version you should use CUDA v10.0 and CUDNN v7.4. Probably, you are not using a valid CUDNN version.

Also, remember adding the environment variables in Windows:
\# SET PATH=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0\bin;%PATH%
\# SET PATH=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0\extras\CUPTI\libx64;%PATH%
\# SET PATH=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0\include;%PATH%
\# SET PATH=C:\tools\cuda\bin;%PATH%

Regards,
José

**Comment by 179416326** [comment]
Hi José, i re-installed CUDNN7.4.2 for testing, but still the demo cannot run. i debuged routenet_with_link_cap.py in pycharm, it showed like this:

Instructions for updating:
Use standard file utilities to get mtimes.
I0326 11:44:22.388325  4452 session_manager.py:500] Running local_init_op.
I0326 11:44:22.478085  4452 session_manager.py:502] Done running local_init_op.
I0326 11:44:28.776959  4452 basic_session_run_hooks.py:606] Saving checkpoints for 0 into ./CheckPoints/nsfnetbw\model.ckpt.
W0326 11:45:50.676099  4452 ops.py:465] Tensor._shape is private, use Tensor.shape instead. Tensor._shape will eventually be removed.
W0326 11:45:50.701011  4452 ops.py:465] Tensor._shape is private, use Tensor.shape instead. Tensor._shape will eventually be removed.

Process finished with exit code -1073741819 (0xC0000005)

**Comment by 179416326** [comment]
In pycharm terminal, i put this:
python routenet_with_link_cap.py train --hparams="l2=0.1,dropout_rate=0.5,link_state_dim=32,path_state_dim=32,readout_units=256,learning_rate=0.001,T=8" --train d:/omnetpp-6.0pre7/samples/routenet/nsfnetbw/tfrecords/train/*.tfrecords --train_steps 50000 --eval_ d:/omnetpp-6.0pre7/samples/routenet/nsfnetbw/tfrecords/evaluate/*.tfrecords --model_dir ./CheckPoints/nsfnetbw

Is this command right? i wonder the running failure is because of the wrong "--train " parameter and "--eval_" parameter  in windows10. That means the demo cannot get "*.tfrecords" from the right file path.

**Comment by jsuarezv** [comment]
Hi Watson,
Yes, the command looks right. I assume that the paths to "train" and "eval_" include a wildcard at the end ("...\*.tfrecords") but it was removed due to the Markdown language.

I would try to execute it from the Windows terminal just to discard it is a problem with PyCharm.

Regards,
José

**Comment by 179416326** [comment]
Hi José, i did some tests for the wildcard("...*.tfrecords") , it turned out to be the problem with Markdown. More over, i changed the  "train" and "eval_" as input parameters($3 and $4) like this:
sh run_routenet.sh train nsfnetbw 50000 d:/tfrecords/train/*.* d:/tfrecords/evaluate/*.*

Also, some changes in run_routenet.sh:
if [[ "$1" = "train" ]]; then

    python routenet_with_link_cap.py train --hparams="l2=0.1,dropout_rate=0.5,link_state_dim=32,path_state_dim=32,readout_units=256,learning_rate=0.001,T=8" --train_steps $3 --train $4  --eval_ $5 --model_dir ./CheckPoints/$2
fi

However, when i checked the file "graph.pbtxt", it could only find one file(maybe the first one) in "nsfnetbw\tfrecords\train", here it is:
  attr {
    key: "value"
    value {
      tensor {
        dtype: DT_STRING
        tensor_shape {
          dim {
            size: 1
          }
        }
        string_val: "d:/tfrecords/train/results_nsfnetbw_10-12_Routing-SP-k-0_0_124.tfrecords"
      }
    }
  }
}
Best regards,
Wayne

**Comment by jsuarezv** [comment]
Hi Wayne,
Please note that this demo is prepared to use the datasets we provide [here](https://github.com/knowledgedefinednetworking/NetworkModelingDatasets/tree/master/datasets_v0) ('datasets_v0'). 
Looking at the names of your tfrecords I see you are using 'datasets_v1', which have a different structure. In this case, you would need to modify the code to read correctly the tfrecords.

Regards,
José

**Comment by 179416326** [comment]
Yes, i did use datasets_v0 for testing. For datasets_v0, i ran it well in Ubuntu, but for the reason of wildcard("/*.tfrecords"), the demo couldnot get the right path for xx.tfrecords files in pycharm terminal(windows cmd). For the testing above, i put "train" and "eval_" as input parameters($4 and $5) trying to solve the problem of wildcard, so i am puzzled why it found only one .tfrecords file in "graph.pbtxt".
